lib/pmd.js

```diff
@@ -280,9 +283,11 @@ pmd.generateData = function(config){
         (folder.source.fileFilterRegexp instanceof RegExp && folder.source.fileFilterRegexp.test(files[i])) ||
         !folder.source.fileFilterRegexp instanceof RegExp){
 
+        let newextname = path.extname(config.folders[0].template);
         file.ext = path.extname(files[i]);
         file.name = path.basename(files[i], file.ext);
         file.path = path.resolve(folder.source.path, files[i]);
+        file.fileName = file.name + newextname;
```

Comment:
From Line `286`:

- [ ] use `docExtName` instead of `newextname`
- [ ] `file.fileName` seems a bit confusing. Perhaps `file.docFileName` or similar.